pkg/deployments/tasks/20210624-stable-pool/test/task.fork.ts

```diff
@@ -0,0 +1,96 @@
+import { getForkedNetwork } from '../../../src/test';
+import { getSigner, impersonateWhale } from '../../../src/signers';
+import { MAX_UINT256 } from '@balancer-labs/v2-helpers/src/constants';
+import { SWAP_KIND } from '@balancer-labs/v2-helpers/src/models/vault/swaps';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+
+describe('StablePool', function () {
+  let owner: SignerWithAddress, whale: SignerWithAddress;
+  let pool: Contract, factory: Contract, vault: Contract, usdc: Contract, dai: Contract;
+
+  const task = Task.forTest('20210624-stable-pool', getForkedNetwork(hre));
+
+  const DAI = '0x6b175474e89094c44da98b954eedeac495271d0f';
+  const USDC = '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48';
+
+  const tokens = [DAI, USDC];
+  const amplificationParameter = bn(100);
+  const swapFeePercentage = fp(0.01);
+  const initialBalanceDAI = fp(1e6);
+  const initialBalanceUSDC = fp(1e6).div(1e12); // 6 digits
+  const initialBalances = [initialBalanceDAI, initialBalanceUSDC];
+
+  before('run task', async () => {
+    await task.run({ force: true });
+    factory = await task.instanceAt('StablePoolFactory', task.output().factory);
+  });
+
+  before('load signers', async () => {
+    owner = await getSigner();
+    whale = await impersonateWhale(fp(100));
+  });
+
+  before('load vault and tokens', async () => {
+    const vaultTask = Task.forTest('20210418-vault', getForkedNetwork(hre));
+    vault = await vaultTask.instanceAt('Vault', await factory.getVault());
+    dai = await task.instanceAt('IERC20', DAI);
+    usdc = await task.instanceAt('IERC20', USDC);
+  });
+
+  it('deploy a stable pool', async () => {
+    const tx = await factory.create('SP', 'SPT', tokens, amplificationParameter, swapFeePercentage, owner.address);
+    const event = expectEvent.inReceipt(await tx.wait(), 'PoolCreated');
+
+    pool = await task.instanceAt('StablePool', event.args.pool);
+    expect(await factory.isPoolFromFactory(pool.address)).to.be.true;
+  });
+
+  it('can initialize a stable pool', async () => {
+    await dai.connect(whale).approve(vault.address, MAX_UINT256);
+    await usdc.connect(whale).approve(vault.address, MAX_UINT256);
+
+    const poolId = await pool.getPoolId();
+    const userData = encodeJoinStablePool({ kind: 'Init', amountsIn: initialBalances });
+    await vault.connect(whale).joinPool(poolId, whale.address, owner.address, {
+      assets: tokens,
+      maxAmountsIn: initialBalances,
+      fromInternalBalance: false,
+      userData,
+    });
+
+    const expectedInvariant = calculateInvariant([initialBalanceDAI, initialBalanceDAI], amplificationParameter);
+    expectEqualWithError(await pool.balanceOf(owner.address), expectedInvariant, 0.001);
+  });
+
+  it('can swap in a stable pool', async () => {
+    const amount = fp(500);
+    await dai.connect(whale).transfer(owner.address, amount);
+    await dai.connect(owner).approve(vault.address, amount);
```

Comment:
Why not do the swap from the whale directly? It feels odd to use the owner for this - I imagined you'd use the whale as an LP (same with the join above)